src/DeterministicNoiseSimulator.cpp

```diff
@@ -2,16 +2,13 @@
 
 #include "dd/Export.hpp"
 
-using CN    = dd::ComplexNumbers;
-using dEdge = dd::dEdge;
-using dNode = dd::dNode;
-using mEdge = dd::mEdge;
+using CN = dd::ComplexNumbers;
 
 template<class DDPackage>
 std::map<std::string, double> DeterministicNoiseSimulator<DDPackage>::DeterministicSimulate() {
     std::map<unsigned int, bool> classicValues;
 
-    rootEdge = Simulator<DDPackage>::dd->makeZeroDensityOperator();
+    rootEdge = Simulator<DDPackage>::dd->makeZeroDensityOperator(this->dd->qubits());
```

Comment:
the size of the density operator is guided by the number of qubits in the circuit and not the number of qubits configured in the DD Package instance. Just because these values happen to be the same in all tested cases, this still is the wrong intent.

Reply:
rootEdge = Simulator<DDPackage>::dd->makeZeroDensityOperator(qc->getNqubits());
> rootEdge = Simulator<DDPackage>::dd->makeZeroDensityOperator(qc->getNqubits());